servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -229,25 +248,25 @@ private void assertInEventloop() {
     }
 
     private final class SrvRecordPublisher extends AbstractDnsPublisher<HostAndPort> {
-        private final String serviceName;
 
-        private SrvRecordPublisher(String serviceName) {
-            this.serviceName = requireNonNull(serviceName);
+        private SrvRecordPublisher(final String serviceName, @Nullable final DnsDiscoveryObserver discoveryObserver) {
+            super(serviceName, discoveryObserver);
         }
 
         @Override
         public String toString() {
-            return serviceName;
+            return "SRV lookups for " + name();
         }
 
         @Override
         protected AbstractDnsSubscription newSubscription(
-                final Subscriber<? super Iterable<ServiceDiscovererEvent<HostAndPort>>> subscriber) {
-            return new AbstractDnsSubscription(true, subscriber) {
+                final Subscriber<? super Iterable<ServiceDiscovererEvent<HostAndPort>>> subscriber,
+                @Nullable final DnsDiscoveryObserver discoveryObserver) {
```

Comment:
Just make the `discoveryObserver` `protected` in the base class instead of passing back again in this method.